### PR TITLE
[VAS-850][VAS-857][VAS-869][VAS-870] feat: add repo skeleton and infra

### PR DESCRIPTION
**Pull request by alessio-cialini**
#### List of Changes

 - Introduced missing maven wrapper
 - Updated actions with project name
 - Updated README.md
 - Introduced infra templates
 - Introduced missing helm
 - Introduced openapi generator
 - Introduced service skeleton
 - generated openapi.json 

#### Motivation and Context

https://pagopa.atlassian.net/jira/software/c/projects/VAS/boards/540?selectedIssue=VAS-857
https://pagopa.atlassian.net/jira/software/c/projects/VAS/boards/540?selectedIssue=VAS-850

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by alessio-cialini** [comment]
update_code

**Comment by alessio-cialini** [comment]
_update_code_

**Review by svariant — APPROVED** [review]
🧑‍💻👍🏻

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    rev: v1.11.0
+    hooks:
+      - id: ggshield
+        language_version: python3
+        stages: [ commit ]
+
+  - repo: https://github.com/maltzj/google-style-precommit-hook
+    rev: b7e9e7fcba4a5aea463e72fe9964c14877bd8130
+    hooks:
+      - id: google-style-java
```
Comment (jacopocarlini) [inline .pre-commit-config.yaml]:
```suggestion
```

File: helm/values-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+microservice-chart:
+  namespace: "gps"
+  nameOverride: ""
+  fullnameOverride: ""
```
Comment (jacopocarlini) [inline helm/values-dev.yaml]:
```suggestion
  fullnameOverride: "gpd-payments-pull"
```

File: helm/values-prod.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
+microservice-chart:
+  namespace: "gps"
+  nameOverride: ""
+  fullnameOverride: ""
```
Comment (jacopocarlini) [inline helm/values-prod.yaml]:
```suggestion
  fullnameOverride: "gpd-payments-pull"
```

File: helm/values-uat.yaml [hunk 4]
```diff
@@ -0,0 +1,118 @@
+microservice-chart:
+  namespace: "gps"
+  nameOverride: ""
+  fullnameOverride: ""
```
Comment (jacopocarlini) [inline helm/values-uat.yaml]:
```suggestion
  fullnameOverride: "gpd-payments-pull"
```

File: openapi/generate_openapi.sh
Comment (jacopocarlini) [inline openapi/generate_openapi.sh:1]: this is an old version. Use the new (you can see backoffice)

File: infra/04_apim_api.tf [hunk 5]
```diff
@@ -0,0 +1,55 @@
+locals {
+  apim_gpd_payments_pull = {
+    // Backoffice External
```
Comment (jacopocarlini) [inline infra/04_apim_api.tf]:
```suggestion
```

File: infra/04_apim_api.tf [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+## OpenApi  ##
+##############
+
+module "apim_gpd_payments_pull_api_v1" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//api_management_api?ref=v6.4.1"
+
+  name                  = format("%s-gpd-payments-pull-api", var.env_short)
+  api_management_name   = local.apim.name
+  resource_group_name   = local.apim.rg
+  product_ids           = [local.apim.bo_external_product_psp_id]
```
Comment (jacopocarlini) [inline infra/04_apim_api.tf]: check the local

File: infra/99_locals.tf [hunk 7]
```diff
@@ -0,0 +1,12 @@
+locals {
+  product = "${var.prefix}-${var.env_short}"
+
+  apim = {
+    name                   = "${local.product}-apim"
+    rg                     = "${local.product}-api-rg"
+    bo_external_product_psp_id = "selfcare-bo-external-psp"
+    bo_external_product_ec_id = "selfcare-bo-external-ec"
+    bo_helpdesk_product_id = "selfcare-bo-helpdesk"
```
Comment (jacopocarlini) [inline infra/99_locals.tf]: use the right product

File: src/main/java/it/gov/pagopa/gpd/payments/pull/models/Installment.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Jacksonized
+@RegisterForReflection
+public class Installment implements Serializable {
+
+    private String nav;
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/gpd/payments/pull/models/Installment.java:24]: add the descriptions for the opeanpi file

File: src/main/java/it/gov/pagopa/gpd/payments/pull/models/PaymentNotice.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Jacksonized
+@RegisterForReflection
+public class PaymentNotice implements Serializable {
+
+    private String iupd;
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/gpd/payments/pull/models/PaymentNotice.java:25]: ditto

**Review by jacopocarlini — CHANGES_REQUESTED** [review]
👍 
see the comments

**Review by jacopocarlini — COMMENTED** [review]
this is a great job! kudos 🚀 🥇

**Review by jacopocarlini — APPROVED** [review]

**Pull request merged**